### ISSUE TITLE
BMD and CEN (patchy implementation) need per-page title elements, as on REG

### ISSUE THREAD

**Issue by AlOneill** [FreeUKGen/FreeBMD2#515]
Although REG sets the `title` element in the `head` on a per page basis, BMD and CEN do not.

The BMD default `title` is misleading as it says, "FreeBMD - UK Birth, Marriage, and Death Records (England, <em>Scotland</em>, Wales)". The emphasis is mine — Scotland records are not included on BMD.

**Comment by AlOneill** [comment]
The contents of the `title` element is the first thing a screen reader user will hear when they reach a page. It helps everyone to know what the page is about, and whether they are in the right place. Sighted users will expect to see that in the browser tab. When a user returns to a browser tab later, they will know which page they are on.

For sighted users with a lot of tabs open, it helps to put the site name last — the tab icon will be enough for many to identify the site. For example, "Accessibility statement | FreeBMD". This is the pattern on REG.

**Comment by DeniseColbert** [comment]
@Vino-S will look at the code for FreeREG and see if it can be taken over to CEN & BMD2

**Comment by Vino-S** [comment]
CEN: Ready for testing in test3 CEN

**Comment by AlOneill** [comment]
@Vino-S looking good, but should be cased as "FreeCEN" rather than "FREECEN" and I'm not sure that app pages have the most useful `title`s — I will need to make a more thorough check.

**Comment by DeniseColbert** [comment]
Vino will fix casing

**Comment by PatReynolds** [comment]
Done now on FreeCEN, but casing needing attention.

**Comment by AlOneill** [comment]
@Vino-S Casing CEN now looks good but am not seeing update on BMD

**Comment by PatReynolds** [comment]
Moved to production, so closing.

**Comment by AlOneill** [comment]
Still not seeing title element contents in browser tab on BMD t3 — re-opening.

**Comment by richardofsussex** [comment]
I'm not sure where this issue has got to, but the fix is to update this code in app/helpers/application_helper.rb (line 369 onwards):
` def project_description
    {
      freereg: "UK Parish Register Records",
      freecen: "UK Census Records (England, Scotland, Wales)",
      freebmd: "UK Birth, Marriage, and Death Records (England, Scotland, Wales)"
    }
  end`

**Comment by AlOneill** [comment]
@richardofsussex For FreeBMD, we do not have records for Scotland, but you knew that already :-), so:

`def project_description { freereg: "UK Parish Register Records", freecen: "UK Census Records (England, Scotland, Wales)", freebmd: "UK Birth, Marriage, and Death Records (England, Wales)" } end`

or something similar.

In addition, pages other than the search page will need their own version of the `title` text as used for FreeREG pages.

**Comment by AlOneill** [comment]
There remains an issue with some of the `title` elements: whilst doing something else I noticed that the page I was looking at had `<title>String | BMD</title>` — oops.

The `h1` for the page was, "Entries in Page: 288, Volume: 8 at District: Wimborne for Apr to Jun 1848"

**Comment by richardofsussex** [comment]
I fixed this by simply adding <% title "Same page entries" %> to same_page_entries.html.erb. @Vino-S please merge the fb_473_rbl branch into test3 so @AlOneill can double-check.

Any other pages you find with a title "String | BMD" can be fixed in the same way.

**Comment by DeniseColbert** [comment]
@Vino-S please check Richard's comment and if it's right, push into test3 so @AlOneill can check

**Comment by DeniseColbert** [comment]
@AlOneill to test on test3

**Comment by DeniseColbert** [comment]
@AlOneill is this done now?

**Comment by AlOneill** [comment]
@DeniseColbert 
I do not completely understand how the contents of each title element is set up for the app pages — I hope that it can be set up to be automatic because we do not use a distinct page setup for each instance but rather a template, if you see what I mean? However, I can let the developers know of pages that I come across that do not have a fully-formed `title` element and maybe they can look at the pages themselves and devise a fix for any that are incomplete?

**For instance**, I checked the Records page: it is incomplete as it reads "FreeBMD |" whereas a complete `title` for the top-level Records page would be "Records | FreeBMD".

On selecting "B" I would hope for a title of "Districts starting with B | FreeBMD"; then on selecting Banbury, the `title` would be "Banbury District Information | FreeBMD". In other words, the `title` should be specific to the page — it cannot be "Records | FreeBMD" for every page in the Records section, however tempting that might be!

TBH, I would have to check, but I do not believe that the `title` elements are properly specific on FreeREG or FreeCEN. We need to fix them too. Sorry, developers if this looks like a faff, but it is important (even essential) for some of our users and helpful for all our users.

**Comment by AlOneill** [comment]
@DeniseColbert 
REG title elements situation is looking better than I expected, but there are still some oddities. (I have run out of steam.)

**Comment by DeniseColbert** [comment]
Is BMD in line with REG? Probably not, but let's find out what's needed.

**Comment by DeniseColbert** [comment]
@AlOneill please test on test3 when Vino updates the comments with the changes she has made

**Comment by richardofsussex** [comment]
I have added per-page titles for Districts and Entry information pages, plus titles for Help and About. Results on beta

**Comment by AlOneill** [comment]
The `title` element for the initial Database page on beta is "FreeBMD |" which is not quite right. There is nothing specific for County either.

**Comment by AlOneill** [comment]
Someone needs to check all the other pages on the site that are public-facing, i.e. that a member of the public could navigate to.

**Comment by richardofsussex** [comment]
I have made an attempt to do this: see branch `fb_515_rbl`. @Vino-S will have to merge this code into test3 before you will see any differences. She is on holiday this week.

**Comment by KristinaGadzhieva** [comment]
Done for Database page, @Vino-S please check other pages

**Comment by richardofsussex** [comment]
Problem tends to be the use of the HTML `<title>` element, rather than assigning a string to the Rails title variable.

**Comment by richardofsussex** [comment]
I have been through all the pages I can identify, adding suitable `title` values to them. Branch `fb_515_rbl_2` - yet to be uploaded to test3.

**Comment by Vino-S** [comment]
Deployed

**Comment by DeniseColbert** [comment]
Deployed: closing
